source/inject.ts

```diff
@@ -22,88 +17,118 @@ const injectAndDiscardCertainErrors: typeof injectContentScript = async (tabId,
 	}
 };
 
-function forgetTab(tabId: number) {
-	tracked.delete(tabId);
-	if (tracked.size === 0) {
-		chrome.tabs.onUpdated.removeListener(onDiscarded);
+const storageKeyRoot = 'webext-inject-on-install:';
+
+function getTabStorageKey(tabId: number) {
+	return `${storageKeyRoot}${tabId}`;
+}
+
+async function forgetTab(tabId: number) {
+	await chrome.storage.session.remove(getTabStorageKey(tabId));
+	const fullStorage = await chrome.storage.session.getKeys();
```

Comment:
It probably won't make much of a difference in practice, but you could put the check for no remaining tabs in requestIdleCallback